substrate/frame/bounties/src/benchmarking.rs

```diff
@@ -17,87 +17,163 @@
+	bounty_id: BountyIndex,
+	caller: T::AccountId,
+) -> Result<(), BenchmarkError> {
 	let approve_origin =
 		T::SpendOrigin::try_successful_origin().map_err(|_| BenchmarkError::Weightless)?;
 	Bounties::<T, I>::approve_bounty(approve_origin.clone(), bounty_id)?;
-	set_block_number::<T, I>(T::SpendPeriod::get());
-	Treasury::<T, I>::on_initialize(frame_system::Pallet::<T>::block_number());
-	Bounties::<T, I>::propose_curator(approve_origin, bounty_id, curator_lookup.clone(), fee)?;
-	Bounties::<T, I>::accept_curator(RawOrigin::Signed(curator).into(), bounty_id)?;
-	Ok((curator_lookup, bounty_id))
+
+	approve_last_payment();
+	Ok(())
+}
+
+fn approve_bounty_and_propose_curator<T: Config<I>, I: 'static>(
+) -> Result<(T::AccountId, BountyIndex), BenchmarkError> {
+	let (caller, curator, _asset_kind, fee, _value, curator_stash, _reason) =
+		setup_bounty::<T, I>(0, T::MaximumReasonLength::get());
+
+	let bounty_id = create_proposed_bounty::<T, I>()?;
+	initialize_approved_bounty::<T, I>(bounty_id, caller.clone())?;
+	Bounties::<T, I>::check_payment_status(RawOrigin::Signed(caller).into(), bounty_id)?;
+
+	let curator_lookup = T::Lookup::unlookup(curator.clone());
+	Bounties::<T, I>::propose_curator(
+		T::SpendOrigin::try_successful_origin().map_err(|_| BenchmarkError::Weightless)?,
+		bounty_id,
+		curator_lookup.clone(),
+		fee,
+	)?;
+	Ok((curator, bounty_id))
+}
+
+fn approve_bounty_and_accept_curator<T: Config<I>, I: 'static>(
+) -> Result<(T::AccountId, BountyIndex), BenchmarkError> {
+	let (_caller, _curator, _asset_kind, _fee, _value, curator_stash, _reason) =
+		setup_bounty::<T, I>(0, T::MaximumReasonLength::get());
+	let (curator, bounty_id) = approve_bounty_and_propose_curator::<T, I>()?;
+	let curator_stash_lookup = T::BeneficiaryLookup::unlookup(curator_stash);
+	Bounties::<T, I>::accept_curator(
+		RawOrigin::Signed(curator.clone()).into(),
+		bounty_id,
+		curator_stash_lookup,
+	)?;
+	Ok((curator, bounty_id))
+}
+
+fn create_curator_and_award_bounty<T: Config<I>, I: 'static>(
+) -> Result<(T::AccountId, BountyIndex), BenchmarkError> {
+	let (_caller, curator, _asset_kind, _fee, _value, curator_stash, _reason) =
+		setup_bounty::<T, I>(0, T::MaximumReasonLength::get());
+	let (curator, bounty_id) = approve_bounty_and_accept_curator::<T, I>()?;
+	let beneficiary_account: T::Beneficiary = account("beneficiary", 0, SEED);
+	let beneficiary = T::BeneficiaryLookup::unlookup(beneficiary_account.clone());
+
+	Bounties::<T, I>::award_bounty(
+		RawOrigin::Signed(curator.clone()).into(),
+		bounty_id,
+		beneficiary.clone(),
+	)?;
+
+	set_block_number::<T, I>(
+		T::SpendPeriod::get() + T::BountyDepositPayoutDelay::get() + 1u32.into(),
+	);
+
+	Ok((curator, bounty_id))
 }
 
 fn setup_pot_account<T: Config<I>, I: 'static>() {
-	let pot_account = Bounties::<T, I>::account_id();
-	let value = minimum_balance::<T, I>().saturating_mul(1_000_000_000u32.into());
+	let pot_account = T::PalletId::get().into_account_truncating();
```

Comment:
use Pallet::account_id()? is it treasury pot? you most probably do not need to fund the treasury pot. the pallet never really transfers the money, but only calls `pay` and `check_status`. so you need to mock them somehow to make the behave the way you need for a certain bench